async.go

```diff
@@ -0,0 +1,136 @@
+package plc
+
+import (
+	"golang.org/x/sync/errgroup"
+)
+
+type asyncer interface {
+	Wait() error
+	Add(name string, value interface{})
+	AddError(err error)
+}
+
+type action func(string, interface{}) error
+
+type job struct {
+	name  string
+	value interface{}
+}
+
+var noJob = job{}
+
+type async struct {
+	action
+	*errgroup.Group
+	jobs        chan job
+	numRoutines int
+}
+
+func newAsync(act action) async {
+	as := async{
+		action:      act,
+		Group:       &errgroup.Group{},
+		jobs:        make(chan job, 1),
+		numRoutines: 1,
+	}
+
+	as.launchRoutine(noJob)
+
+	go func() {
+		// Listen for new jobs and launch them in new goroutines until the limit has been reached.
+		for newJob := range as.jobs {
+			as.numRoutines++
+			as.launchRoutine(newJob)
```

Comment:
Here it probably makes sense to break if numRoutines > maxRoutines to stop spawning processes.